### DISCUSSION TITLE
[v5] how to load the inverse relationship

### DISCUSSION THREAD

**Discussion by brandonsousa** [adonisjs/core#2768] [General]
if to load USER's PROFILE use preload

`User.query()......preload('profile')`

how do I load the PROFILE USER

**Comment by arthurcvm** [comment]
Hi @brandonsousa, send your `user` and `profile` model code.

**Reply by brandonsousa** [reply]
User
```
@hasMany(() => Profile)
public profiles: HasMany<typeof Profile>;
```
	
Profile
```
@belongsTo(() => User)
	public user: BelongsTo<typeof User>;
```

**Reply by arthurcvm** [reply]
In user controller:
```ts
User.query()......preload('profiles')
```
And in profile:
```ts
Profile.query()......preload('user')
```

**Reply by brandonsousa** [reply]
```
const profiles = await Profile.query().where('status', 'isPublished').preload('users');
```
error
E_MISSING_MODEL_ATTRIBUTE: "Profile.user" expects "user_id" to exist on "User" model, but is missing

**Reply by arthurcvm** [reply]
Please send your migrations with relationships definition.

**Reply by brandonsousa** [reply]
table.integer('user_id')
				.notNullable()
				.unsigned()
				.references('id')
				.inTable('users')
				.onUpdate('CASCADE');

**Reply by arthurcvm** [reply]
In Profile model you set this:

```ts
@hasOne(() => User, {
    foreignKey: 'id',
    localKey: 'user_id'
  })
  public user: HasOne<typeof User>
```
and in User model this:

```ts
@belongsTo(() => Profile, {
    foreignKey: 'user_id'
  })
  public profile: BelongsTo<typeof Profile>
```

**Reply by brandonsousa** [reply]
Thanks, it worked, I was trying to use the same logic that was used in v4
